### PR TITLE
Fix typos in docs/spec/api.md

### PR DESCRIPTION
**Pull request by jonjohnsonjr**
Fix grammatical and formatting issues.
Add links to sections or other documents where appropriate.

Signed-off-by: Jon Johnson <jonjohnson@google.com>

### CODE REVIEW

**Comment by aaronlehmann** [comment]
Thanks for the PR. Since `docs/spec/api.md` is an automatically generated file, can you please make these changes in `docs/spec/api.md.tmpl` and run `make docs/spec/api.md` to regenerate it?

**Comment by aaronlehmann** [comment]
Edit: Sorry about the comment above. I didn't notice that api.md.tmpl was modified as well.

File: docs/spec/api.md.tmpl [hunk 1]
```diff
@@ -1003,9 +1008,9 @@ to _b_:
 Link: <<url>?n=2&last=b>; rel="next"
 ```
 
-The client can then issue the request with above value from the `Link` header,
-receiving the values _c_ and _d_. Note that n may change on second to last
-response or be omitted fully, if the server may so choose.
+The client can then issue the request with the above value from the `Link`
+header, receiving the values _c_ and _d_. Note that `n` may change on second
+to last response or be fully omitted, depending on the server implementation.
```
Comment (aaronlehmann) [inline docs/spec/api.md.tmpl]: Should this be "`n` may change on the second to last response"?

Comment (jonjohnsonjr) [inline docs/spec/api.md.tmpl]: Oops, missed that. Thanks.

**Comment by aaronlehmann** [comment]
LGTM

**Pull request merged**